### PR TITLE
Fix/1554 user focus can escape dialog when navigating with tab

### PR DESCRIPTION
**Pull request by anna-lach**
- Implements focus-trap in the `dialog`
- Creates `FocusTrapController`

### CODE REVIEW

**Review by Diaan — APPROVED** [review]

File: packages/components/dialog/src/dialog.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Button } from '@sl-design-system/button';
 import { ButtonBar } from '@sl-design-system/button-bar';
 import { Icon } from '@sl-design-system/icon';
 import { type EventEmitter, breakpoints, event } from '@sl-design-system/shared';
+import { FocusTrap, createFocusTrap } from 'focus-trap';
```
Comment (jpzwarte) [inline packages/components/dialog/src/dialog.ts:7]: We should only import the `FocusTrap` *type* right?

File: packages/components/dialog/src/dialog.ts [hunk 2]
```diff
@@ -65,6 +66,8 @@ export class Dialog extends ScopedElementsMixin(LitElement) {
   /** @internal */
   static override styles: CSSResultGroup = [breakpoints, styles];
 
+  #focusTrap?: FocusTrap;
```
Comment (jpzwarte) [inline packages/components/dialog/src/dialog.ts:69]: How about making this a `FocusTrapController`? And hide the logic inside the controller?

**Review by jpzwarte — APPROVED** [review]

File: packages/components/dialog/src/dialog.ts (hunk 2 above)
Comment (jpzwarte) [reply packages/components/dialog/src/dialog.ts:69]: That way we also don't use `focus-trap` directly *except* inside the controller?

File: packages/components/dialog/src/dialog.ts [hunk 3]
```diff
@@ -94,6 +97,14 @@ export class Dialog extends ScopedElementsMixin(LitElement) {
     super.connectedCallback();
 
     this.inert = true;
+
+    this.#focusTrapController = new FocusTrapController(this, { disableCancel: this.disableCancel });
```
Comment (jpzwarte) [inline packages/components/dialog/src/dialog.ts:101]: Is this necessary in `connectedCallback` perse? Can't you initialize the controller on line 69?

Comment (jpzwarte) [reply packages/components/dialog/src/dialog.ts:101]: You probably also want to update `disableCancel` when that property changes? So add that code to `updated()` lifecycle callback?

File: packages/components/dialog/src/dialog.ts [hunk 4]
```diff
@@ -94,6 +97,14 @@ export class Dialog extends ScopedElementsMixin(LitElement) {
...
+  }
+
+  override disconnectedCallback(): void {
+    super.disconnectedCallback();
+
+    this.#focusTrapController?.deactivate();
```
Comment (jpzwarte) [inline packages/components/dialog/src/dialog.ts:107]: Can't this happen automatically? In the `hostDisconnected` callback in the controller?

**Review by jpzwarte — APPROVED** [review]

**Review by Diaan — APPROVED** [review]

**Pull request merged**